packages/api-client-core/src/InternalModelManager.ts

```diff
@@ -207,9 +225,8 @@ export class InternalModelManager {
   }
 
   async findMany(options?: RecordData, throwOnEmptyData?: boolean, isFirstQuery = false): Promise<GadgetRecordList<any>> {
-    const response = await this.connection.currentClient
-      .query(internalFindManyQuery(this.apiIdentifier, isFirstQuery), options)
-      .toPromise();
+    const plan = isFirstQuery ? internalFindFirstQuery(this.apiIdentifier, options) : internalFindManyQuery(this.apiIdentifier, options);
```

Comment:
If this function is now calling different functions based on `isFirstQuery` or not, is there any point to re-using it from `findFirst`? Should we just make them each call their own query builder? Lucas and I went back and forth on how much to re-use these functions to implement each other, and I like the re-use to avoid bloating the bundle size too much, and/or so we don't need to duplicate too much, but it also seems a bit convoluted at this point.

Reply:
Yeah this could be separated out. Once I refactored the query itself into using objects rather than strings it was becoming harder to read with conditionals inside objects. Other than reading the same `listX` graphql field very little of the query was really shared.

Reply:
I've pushed another commit that breaks it out into it's own operation to get a feel for what it's like. I think it's definitely simpler, it removes the `throwOnEmptyData` option from `findMany` and `isFirstQuery`. This means the combination of options is significantly reduced.